crawlers/page_spider.py

```diff
@@ -92,7 +95,14 @@ def extract_links(self, response):
             process_value=self.config["link_extractor_process_value"],
         )
 
-        urls_found = set(i.url for i in links_extractor.extract_links(response))
+        urls_found = list(set(i.url for i in links_extractor.extract_links(response)))
+        broken_urls = urls_found
+        urls_found = set(filter(lambda url: validators.url(url) == True, urls_found))
+        broken_urls = set(broken_urls) ^ set(urls_found)  # returns the difference between the two lists.
+
+        print(f"+{len(broken_urls)} broken urls found...")
+        print(f"Broken Urls (filtered): {broken_urls}")
```

Comment:
Somente um nitpick aqui. Acho que valia adicionar uma condição aqui pra não printar a lista se não existem urls quebradas, só pra não aparecer `Broken Urls (filtered): {}` em toda execução e deixar o log mais limpo.
```suggestion
        if broken_urls:
            print(f"Broken Urls (filtered): {broken_urls}")
```

Reply:
Corrigido